cpp/src/Ice/Incoming.cpp

```diff
@@ -241,8 +223,22 @@ IceInternal::IncomingBase::warning(const Exception& ex) const
 }
 
 void
-IceInternal::IncomingBase::warning(const string& msg) const
+IceInternal::IncomingBase::warning(std::exception_ptr ex) const
 {
+    string msg;
```

Comment:
I would move `Warning out` declaration here, and use it directly instead of creating `msg`.

Reply:
Fixed as suggested.